Sources/momento/CacheClient.swift

```diff
@@ -0,0 +1,300 @@
+        // handle error
+     case let responseSuccess as CacheSetSuccess:
+        // handle success
+     }
+    ```
+     */
+    public func set(cacheName: String, key: Data, value: String, ttl: TimeInterval? = nil) async -> CacheSetResponse {
+        do {
+            try validateCacheName(cacheName: cacheName)
+            try validateCacheKey(key: key)
+            try validateCacheValue(value: value)
+            try validateTtl(ttl: ttl)
+        } catch let err as SdkError {
+            return CacheSetError(error: err)
+        } catch {
+            return CacheSetError(error: UnknownError(
+                message: "unexpected error: \(error)")
+            )
+        }
+        return await self.dataClient.set(cacheName: cacheName, key: key, value: value, ttl: ttl)
+    }
+    
+    /**
+     Associates the given key with the given value. If a value for the key is already present it is replaced with the new value.
+     - Parameters:
+        - cacheName: the name of the cache to store the value in
+        - key: the key to set the value for
+        - value: the value to associate with the key
+     - Returns: CacheSetResponse representing the result of the set operation.
+     Pattern matching can be used to operate on the appropriate subtype.
+    ```
+     switch response {
+     case let responseError as CacheSetError:
+        // handle error
+     case let responseSuccess as CacheSetSuccess:
+        // handle success
+     }
+    ```
+     */
+    public func set(cacheName: String, key: String, value: Data, ttl: TimeInterval? = nil) async -> CacheSetResponse {
+        do {
+            try validateCacheName(cacheName: cacheName)
+            try validateCacheKey(key: key)
+            try validateCacheValue(value: value)
+            try validateTtl(ttl: ttl)
+        } catch let err as SdkError {
+            return CacheSetError(error: err)
+        } catch {
+            return CacheSetError(error: UnknownError(
+                message: "unexpected error: \(error)")
+            )
+        }
+        return await self.dataClient.set(cacheName: cacheName, key: key, value: value, ttl: ttl)
+    }
+    
+    /**
+     Associates the given key with the given value. If a value for the key is already present it is replaced with the new value.
+     - Parameters:
+        - cacheName: the name of the cache to store the value in
+        - key: the key to set the value for
+        - value: the value to associate with the key
+     - Returns: CacheSetResponse representing the result of the set operation.
+     Pattern matching can be used to operate on the appropriate subtype.
+    ```
+     switch response {
+     case let responseError as CacheSetError:
+        // handle error
+     case let responseSuccess as CacheSetSuccess:
+        // handle success
+     }
+    ```
+     */
+    public func set(cacheName: String, key: Data, value: Data, ttl: TimeInterval? = nil) async -> CacheSetResponse {
```

Comment:
Just an observation I don't have any background in swift, but if its possible to have something like `key: Data | String` then we could reduce alot of the copy pasta'd code between all of these methods

Reply:
That's a good point. Swift uses [enums for type unions](https://camlittle.com/posts/type-unions-in-swift/), I can try that approach out and reduce the amount of copy pasta